build/build.go

```diff
@@ -77,10 +73,10 @@ func (l *Lifecycle) Execute(ctx context.Context, opts LifecycleOptions) error {
 		l.logger.Debugf("Build cache %s cleared", style.Symbol(buildCache.Name()))
 	}
 
-	lifecycleVersion := l.builder.GetLifecycleVersion()
+	lifecycleVersion := l.builder.GetLifecycleDescriptor().Info.Version
 	if lifecycleVersion == nil {
```

Comment:
I think `builder.GetLifecycleDescriptor()` already returns the assumed version

Reply:
It depends greatly on the code path in which I chose not to worry about the probabilities and instead make the code robust.